packages/core/src/routes/experience/classes/verifications/verification-records-map.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * @overview Since map in TS does not support strict key value type checking, we have to manually define the key value type.
+ * This class is used to store and manage all the verification records.
+ * We override the set and get methods to ensure the type safety of the verification record.
+ */
+
+import { type VerificationType } from '@logto/schemas';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import assertThat from '#src/utils/assert-that.js';
+
+import { type VerificationRecord, type VerificationRecordMap } from './index.js';
+
+type VerificationRecordKey<T extends VerificationRecord> = T['type'];
+
+export class VerificationRecordsMap {
```

Comment:
if the whole point of the class is just for typing, then no need to do it in a class since it has no functionality difference

Reply:
updated